pkg/postgres/database.go

```diff
@@ -0,0 +1,75 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/lib/pq"
+)
+
+// Credentials represents connection credentials for a user on a
+// PostgreSQL instance.
+type Credentials struct {
+	Name     string
+	Password string
+}
+
+func Database(log logr.Logger, db *sql.DB, credentials Credentials) error {
+	// Create the service user
+	_, err := db.Exec(fmt.Sprintf("CREATE USER %s WITH PASSWORD '%s' NOCREATEROLE VALID UNTIL 'infinity'", credentials.Name, credentials.Password))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_object" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Service user; %s already exists", credentials.Name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Service user; %s created", credentials.Name))
+	}
+
+	// Create the database
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", credentials.Name))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_database" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Database; %s already exists", credentials.Name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Database; %s created", credentials.Name))
+	}
+
+	// Alter ownership of the database to the database user
+	_, err = db.Exec(fmt.Sprintf("ALTER DATABASE %s OWNER TO %s", credentials.Name, credentials.Name))
+	if err != nil {
+		return err
+	}
+
+	serviceConnection, err := Connect(log, fmt.Sprintf("postgresql://%s:%s@localhost:5432/%s?sslmode=disable", credentials.Name, credentials.Password, credentials.Name))
```

Comment:
Maybe a comment here as to why we connect as the service.
Also, the same thing regarding sslmode

Reply:
Good point 👍 
```suggestion
	// Connect with the newly created role to create the schema with that role. This ensures
	// that the object is in fact owned by the service and not the creator role.
	serviceConnection, err := Connect(log, fmt.Sprintf("postgresql://%s:%s@localhost:5432/%s?sslmode=disable", credentials.Name, credentials.Password, credentials.Name))
```